### PR TITLE
MAT-6189:  Set all Patient reference fields on test case

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-6189](https://jira.cms.gov/browse/MAT-6189)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java [hunk 1]
```diff
@@ -309,6 +309,9 @@ public TestCase updateTestCase(
       validatedTestCase.setJson(
           QiCoreJsonUtil.updateResourceFullUrls(validatedTestCase, madieJsonResourcesBaseUri));
     }
+    validatedTestCase.setJson(
+        QiCoreJsonUtil.replacePatientRefs(
+            validatedTestCase.getJson(), validatedTestCase.getPatientId().toString()));
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseService.java]: I think this should be conditional just for QICore.

File: src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java [hunk 2]
```diff
@@ -316,6 +316,8 @@ public static String updateResourceFullUrls(TestCase testCase, String madieJsonR
     return testCase.getJson();
   }
 
+  // update full urls for non-patient resources
+
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java]: is this intentional? the comment does not match the description of the method below it if that's what the comment is for.

Comment (ethankaplan) [reply src/main/java/cms/gov/madie/measure/utils/QiCoreJsonUtil.java]: it was not!

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java (hunk 1 above)
Comment (ethankaplan) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java]: moved into the bracket above!

**Review by adongare — APPROVED** [review]

**Pull request merged**